engine/crates/engine-v2/src/sources/graphql/deserialize/entities.rs

```diff
@@ -99,12 +103,23 @@ where
         formatter.write_str("a non null entities list")
     }
 
-    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    fn visit_seq<A>(mut self, mut seq: A) -> Result<Self::Value, A::Error>
     where
         A: SeqAccess<'de>,
     {
         while let Some(seed) = self.response_part.next_seed(self.plan) {
-            match seq.next_element_seed(seed) {
+            let next_cache_entry = self.cache_entries.as_mut().and_then(Iterator::next);
```

Comment:
this could be actually just expressed as:
```rust
while let Some(cache_entry) = ... { 
  // deserialize
}
while let Some(seed) = ... {
   // process as before
,}
```